player/lua/select.lua

```diff
@@ -353,6 +359,109 @@ mp.add_key_binding(nil, "select-audio-device", function ()
     })
 end)
 
+local function get_history_entry(line, seen, error_message)
+    local entry = utils.parse_json(line)
+
+    if not entry then
+        mp.msg.warn(error_message)
+        return
+    end
+
+    local time, path, title = unpack(entry)
+
+    if not path then
+        mp.msg.warn(error_message)
+        return
+    end
+
+    if seen[path] then
+        return
+    end
+
+    seen[path] = true
+
+    local status, date = pcall(os.date, options.history_date_format, time)
+
+    if not status then
+        -- Lua 5.1 and LuaJIT return invalid formats as is, but Lua 5.2+ errors,
+        -- so also print the returned error message for that case.
+        mp.msg.warn(error_message .. " - " .. date)
+        return
+    end
+
+    return {
+        date = date,
+        path = path,
+        title = title,
+    }
+end
+
+
+local function format_history_entry(entry, osd_playlist_entry)
+    if entry.title and osd_playlist_entry == "title" then
+        return entry.title
+    end
+
+    if entry.title and osd_playlist_entry == "both" then
+        return entry.title .. " (" .. entry.path .. ")"
+    end
+
+    return entry.path
```

Comment:
> If the media-title is not available, it will display only the filename.

Reply:
Act like `both` always if that an issue. titles are also often completely non informative.

Reply:
`both` doesn't show paths either?

Reply:
How is both not showing path?

Reply:
`both` show filenames, wasn't your quote referring to the fact that this shows paths instead of filenames?

Reply:
`return entry.title .. " (" .. entry.path .. ")"` shows title and path

Reply:
I mean in show-text and select-playlist `both` shows filenames, while here I made it show the path because you usually have files from completely different directories unlike in those case so the full path is needed to understand the history.

Reply:
If you say that a full path is required to understand the history it should always be shown.

`<title|filename> (path)`